be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -346,64 +319,103 @@ StatusOr<ChunkPtr> SpillableHashJoinProbeOperator::pull_chunk(RuntimeState* stat
         }
 #endif
     }
+}
 
-    if (_processing_partitions.empty()) {
-        _acquire_next_partitions();
+Status SpillableHashJoinProbeOperator::_restore_probe_partition(RuntimeState* state) {
+    for (size_t i = 0; i < _probers.size(); ++i) {
+        // probe partition has been processed
+        if (_probe_read_eofs[i]) continue;
+        RETURN_IF_ERROR(_current_reader[i]->trigger_restore(
+                state, *_executor,
+                spill::ResourceMemTrackerGuard(tls_mem_tracker, state->query_ctx()->weak_from_this(),
+                                               std::weak_ptr(_current_reader[i]))));
+        if (_current_reader[i]->has_output_data()) {
+            auto chunk_st = _current_reader[i]->restore(
+                    state, *_executor,
+                    spill::ResourceMemTrackerGuard(tls_mem_tracker, state->query_ctx()->weak_from_this(),
+                                                   std::weak_ptr(_current_reader[i])));
+            if (chunk_st.ok() && chunk_st.value() && !chunk_st.value()->is_empty()) {
+                _probers[i]->push_probe_chunk(state, std::move(chunk_st.value()));
+            } else if (chunk_st.status().is_end_of_file()) {
+                _probe_read_eofs[i] = true;
+            } else if (!chunk_st.ok()) {
+                return chunk_st.status();
+            }
+        }
     }
+    return Status::OK();
+}
 
-    if (!_all_loaded_partition_data_ready()) {
-        RETURN_IF_ERROR(_load_all_partition_build_side(state));
+StatusOr<ChunkPtr> SpillableHashJoinProbeOperator::pull_chunk(RuntimeState* state) {
+    RETURN_IF_ERROR(_status());
+    if (spill_strategy() == spill::SpillStrategy::NO_SPILL) {
+        return HashJoinProbeOperator::pull_chunk(state);
     }
 
-    if (!_all_loaded_partition_data_ready()) {
-        // wait loading partition
-        return nullptr;
-    }
+    _check_partitions();
 
-    if (_current_reader.empty() && _is_finishing && all_probe_partition_is_empty()) {
-        // load data from spiller
+    auto all_probe_partition_is_empty = [this]() {
+        for (auto& _prober : _probers) {
+            if (!_prober->probe_chunk_empty()) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    bool probe_has_output = all_probe_partition_is_empty() && !_has_probe_remain;
```

Comment:
is this condition valid?
```  
bool probe_has_output = !all_probe_partition_is_empty() ||  !_has_probe_remain;
```

Reply:
yes. we have test it.
if any probe partition is not empty, it means we can get data by probe.